vsintegration/src/FSharp.Editor/Diagnostics/DocumentDiagnosticAnalyzer.fs

```diff
@@ -74,20 +74,17 @@ type internal FSharpDocumentDiagnosticAnalyzer [<ImportingConstructor>] () =
             let! sourceText = document.GetTextAsync(ct)
             let filePath = document.FilePath
 
-            let! errors =
-                cancellableTask {
-                    match diagnosticType with
-                    | DiagnosticsType.Semantic ->
-                        let! _, checkResults = document.GetFSharpParseAndCheckResultsAsync("GetDiagnostics")
-                        // In order to eliminate duplicates, we should not return parse errors here because they are returned by `AnalyzeSyntaxAsync` method.
-                        let allDiagnostics = HashSet(checkResults.Diagnostics, diagnosticEqualityComparer)
-                        allDiagnostics.ExceptWith(parseResults.Diagnostics)
-                        return Seq.toArray allDiagnostics
-                    | DiagnosticsType.Syntax -> return parseResults.Diagnostics
-                }
+            let errors =
+                HashSet<FSharpDiagnostic>(parseResults.Diagnostics, diagnosticEqualityComparer)
+
+            if diagnosticType = DiagnosticsType.Semantic then
+                let! _, checkResults = document.GetFSharpParseAndCheckResultsAsync("GetDiagnostics")
+
+                for diagnostic in checkResults.Diagnostics do
+                    errors.Add(diagnostic) |> ignore // We don't care about results and if it's already here
```

Comment:
Will the diags by default compare in the way we want / they used to?

Reply:
Comparer is still the same in the hashset

Reply:
Previously parsing errors were removed now they're included, is that what we want?

Reply:
No, fixed that already